### ISSUE TITLE
JOSS Review Thread

### ISSUE THREAD

**Issue by mbarall** [SCECcode/pycsep#163]
1)  I am one of the reviewers assigned by the Journal of Open Source Software. I will use this issue to discuss my experiences with pyCSEP.

I have been aware of CSEP for a long time, but I have never used pyCSEP, and I have never used Python, so I can provide something of a newcomer's perspective.

**Comment by mbarall** [comment]
The documentation website is very good. But an essential part of the documentation is paywalled.

The statistical tests used by pyCSEP are defined in six journal articles listed on the references page. It is not possible to understand pyCSEP or use it effectively without knowing what the statistical tests are. All six papers are behind paywalls, and purchasing them individually would cost over $200. (I acknowledge that one of the papers is freely available on Zenodo, and that many likely users will have access to the articles through their institution's library.)

I recommend that you post a freely-available document on your website that defines the statistical tests implemented in pyCSEP, including the relevant mathematical formulas.

(I note that such a document is mentioned in the future development roadmap.)

**Comment by mbarall** [comment]
Installation went smoothly.

I created a virtual machine in the cloud, and installed remote desktop software. Then I followed the instructions for installing pyCSEP from source. I installed Miniconda, and then ran the commands shown on the installation page. Everything went smoothly, and all the dependencies were installed automatically.

Good job on the installation.

**Comment by mbarall** [comment]
My first attempt to use pyCSEP was to run the example in the README file to download and plot an earthquake catalog from Comcat. I expected it to produce a plot, but no plot appeared. As a first-time Python user, I did not know why there was no plot, if there was something wrong, or what to do about it. I later tried the tutorial to plot a gridded forecast, and again no plot appeared.

Eventually, with the help of some Internet searches, I figured out how to display plots. This might have been obvious to an experienced user, but for me as a first-time user it was frustrating.

I recommend that you amend the tutorials and README to include commands that actually display the plots on-screen. This would provide a better experience for first-time users.

**Comment by mbarall** [comment]
I have now run all the tutorials. They all work as expected, except that in most cases they don't include the necessary commands to make plots appear on the screen. They also seem like a good introduction to the capabilities of pyCSEP.

**Comment by mbarall** [comment]
The JOSS paper is concise and well-written. Good job.

I recommend that in the summary section, which is intended for a non-specialist audience, you delete the sentence "pyCSEP can evaluate earthquake forecasts expressed as expected rates in space-magnitude bins, and simulation-based forecasts that produce thousands of synthetic seismicity catalogs." That is technical information which belongs in the pyCSEP overview section, and indeed, the overview section already contains that information.

**Comment by kbarnhart** [comment]
@mbarall thanks for these comments. I'm linking this comment to the JOSS review issue: openjournals/joss-reviews/issues/3658

I don't know whether JOSS has a policy regarding supporting information behind paywalls or not (as you bring up above). I'll reach out to the editorial board and see if there is any specific guidance on that point I can provide.

**Comment by mbarall** [comment]
The API documentation looks good. It is a comprehensive description of the classes and functions available in pyCSEP. Although many of the descriptions are quite short, there is enough information so that someone who wants to use pyCSEP in their own project can do so.

**Comment by mbarall** [comment]
I installed pytest and then executed the run_tests script. It performed a series of 127 tests of the code, and all the tests passed. Good work.

**Comment by kbarnhart** [comment]
@wsavran, @mbarall A bit of guidance regarding the level of supporting/theory in the documentation based on my conversation with others on the JOSS board. 

Project documentation should include enough theory that the user can both run and understand what the software is doing and why. The theory should support user choices (e.g., categorical variable choices like which of a set of options to use and continuous variable choices like what value to give for a parameter). 

A point that came up in the editorial board discussion that I want to add as well: while a description of the the theory should be added, it is up to you to determine the extent of rationale, validation, and implications.

E.g., provide description of tests described in Savran et al. (2020) as referenced in the docs [here](https://docs.cseptesting.org/concepts/evaluations.html#catalog-based-forecast-evaluations)
https://pubs.geoscienceworld.org/ssa/bssa/article-abstract/110/4/1799/588154/Pseudoprospective-Evaluation-of-UCERF3-ETAS

I don't suspect this will be too onerous to address. Please let me know if you have questions and/or clarifications.

**Comment by wsavran** [comment]
@mbarall thank you for the review of the code and documentation. i will add technical notes to the documentation that explain the theory behind the tests. i agree that this would stengthen the documentation quite a bit. ill respond here once i have those changes made.

**Comment by kbarnhart** [comment]
@kbarnhart the new docs look great. While looking them over I noticed that there are some heading formatting issues (e.g., CL-test not a heading). Obviously a minor concern relative to the content, but I'd recommend addressing them because they influence the left hand column navigation.

**Comment by wsavran** [comment]
@kbarnhart i agree. thanks for catching that! i pushed out a fix for the bugged up header.

**Comment by mbarall** [comment]
@wsavran Thank you for the new documentation on the theory of CSEP tests.  It's great to have descriptions of the tests all together in one place, and I appreciate that you also included code to invoke each test.

I re-ran all the tutorials, and verified that they all display plots now when run from command line

One quibble: In README.md, the example to download and plot an earthquake catalog still does not display the plot when run from command line, at least not for me.

**Comment by wsavran** [comment]
@mbarall Thanks for looking through everything! I just updated the snippet in README.md to show the plot from the command line. Thanks for pointing that out!

**Comment by mbarall** [comment]
@wsavran Thank you for updating the README.md snippet.  It works for me now.

Feel free to close this issue at your discretion.
